chatbot-core/backend/app/databases/qdrant.py

```diff
@@ -0,0 +1,70 @@
+from qdrant_client import QdrantClient, models
+
+from app.settings import Secrets
+from app.utils.logger import LoggerFactory
+
+logger = LoggerFactory().get_logger(__name__)
+
+
+class QdrantConnector:
+    """
+    Qdrant connector class
+
+    Pattern: Singleton
+    Purpose: Create a single instance of the vector database connection
+    """
+
+    _instance = None
+
+    @classmethod
+    def get_instance(cls) -> QdrantClient:
+        """
+        Get the instance of the vector database connection
+        """
+        if cls._instance is None:
+            cls._instance = cls()._create_qdrant_client()
+
+        return cls._instance
+
+    @classmethod
+    def _create_qdrant_client(cls) -> QdrantClient | None:
+        """
+        Create the vector database connection if there is no any existing connection
+        """
+        try:
+            return QdrantClient(host=Secrets.QDRANT_HOST, port=Secrets.QDRANT_PORT)
+        except Exception as e:
+            logger.error(f"Error initializing vector database: {e}")
+            raise
+
+    def _create_collection(
+        self,
+        collection_name: str,
+        vector_size: int = 1536,
```

Comment:
Why dimensions 1536? It should take default from `Constants.DIMENSIONS` (currently 768).

Reply:
Implemented. In testing progress ...